reporter/src/main/kotlin/reporters/AbstractNoticeReporter.kt

```diff
@@ -49,7 +50,7 @@ abstract class AbstractNoticeReporter : Reporter() {
 
     data class NoticeReport(
             val headers: List<String>,
-            val findings: LicenseFindingsMap,
+            val findings: Map<String, List<String>>,
```

Comment:
I think that 'List<String>' is cleaner since the notice reporter should not care whether the data structure is self sorting or not and also it should not care about duplicates because it now delegates duplicate removal (and ordering) to the component I provided. That component is a mapping from Collection<String>  -> List<String> where I can see nothing wrong with that because it is as generic as possible and as specific as needed. Anyhow this isn't the real reason yet but just a side note.

If I'd keep on using a sorted set I would need to provide a comparator to the sorted set.
This comparator  would need to have information from the processing which is not directly available from those processed statements. So comparator handling would add complexity in.

One might argue that all information needed  for sorting should be exposed, I've decided against that and for the encapsulation since the 'Parts' class might not be clear instantly and in particular also we're dealing with mixed types, e.g. unprocessed entries. This option would also lead to less complexity being hidden.

Even if we'd provide a type suitable for sorting or a sufficiently complex comparator then
there is still the code in the notice reporter which creates new sorted sets which would also need to take care of the comparator handling.

All in all I think dealing with lists is superrior in terms of simplicity.